.github/workflows/dev-deploy.yml

```diff
@@ -0,0 +1,174 @@
+    #if: github.ref == 'refs/heads/main'
+    needs: [build-and-package-for-dev]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: aws-actions/setup-sam@v2
+        with:
+          use-installer: true
+      - uses: actions/download-artifact@v3
+        with:
+          name: packaged-testing.yaml
+
+      - name: Assume the testing pipeline user role
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-region: ${{ env.TESTING_REGION }}
+          role-to-assume: ${{ env.TESTING_PIPELINE_EXECUTION_ROLE }}
+          role-session-name: testing-deployment
+          role-duration-seconds: 3600
+          role-skip-session-tagging: true
+
+      - name: Deploy to testing account
+        run: |
+          sam deploy --stack-name ${TESTING_STACK_NAME} \
+            --template packaged-testing.yaml \
+            --capabilities CAPABILITY_IAM \
+            --region ${TESTING_REGION} \
+            --s3-bucket ${{ secrets.TESTING_ARTIFACTS_BUCKET }} \
+            --image-repository ${{ secrets.TESTING_IMAGE_REPOSITORY }} \
+            --no-fail-on-empty-changeset \
+            --role-arn ${{ env.TESTING_CLOUDFORMATION_EXECUTION_ROLE }}
+
+      - name: create-json
+        uses: jsdaniell/create-json@1.1.2
+        with:
+          name: "notification.json"
+          json: '{"LambdaFunctionConfigurations": [{"Id": "EncryptPDFEventConfiguration","LambdaFunctionArn": "arn:aws:lambda:${{ env.TESTING_REGION }}:${{ secrets.AWS_ACCOUNT_ID }}:function:${{ env.TESTING_STACK_NAME }}-EncryptPDF","Events": ["s3:ObjectCreated:Put"]}]}'
+
+
+      - name: Give Lambda Function access rights to source S3 bucket
+        run: |
+          (aws lambda add-permission --function-name ${TESTING_STACK_NAME}-EncryptPDF \
+            --principal s3.amazonaws.com --statement-id s3invoke --action "lambda:InvokeFunction" \
+            --source-arn arn:aws:s3:::${TESTING_STACK_NAME}-mocho \
+            --source-account ${{ secrets.AWS_ACCOUNT_ID }}) || \
+            (echo "Permission already added")
+
+      - name: Configure notification by S3 bucket to Lambda Function
+        run: |
+          aws s3api put-bucket-notification-configuration --bucket ${TESTING_STACK_NAME}-mocho \
+            --notification-configuration file://notification.json
+
+
+
+  test-lambda-deployment:
+    needs: [deploy-testing]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: install-aws-cli
+        uses: unfor19/install-aws-cli-action@v1
+ 
+      - name: Assume the testing pipeline user role
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-region: ${{ env.TESTING_REGION }}
+          role-to-assume: ${{ env.TESTING_PIPELINE_EXECUTION_ROLE }}
+          role-session-name: testing-deployment
+          role-duration-seconds: 3600
+          role-skip-session-tagging: true
+
+      - name: Test lambda function
```

Comment:
@javanlth testing shouldn't be here. If This pipeline is for deploying, then it should stick to deployment related tasks.

Reply:
So I should create another pipeline to test the function after deployment?